rs/sns/governance/proto/ic_sns_governance/pb/v1/governance.proto

```diff
@@ -2344,3 +2347,22 @@ message Account {
   // subaccount (all bytes set to 0) is used.
   optional Subaccount subaccount = 2;
 }
+
+enum Topic {
+  // Unused, here for PB lint purposes.
+  TOPIC_UNSPECIFIED = 0;
+  // Proposals to set the direction of the DAO by tokenomics & branding
+  TOPIC_DAO_COMMUNITY_SETTINGS = 1;
+  // Proposals to upgrade and manage the SNS DAO framework
+  TOPIC_SNS_FRAMEWORK_MANAGEMENT = 2;
+  // Proposals to manage the dapp's canisters
+  TOPIC_DAPP_CANISTER_MANAGEMENT = 3;
+  // Proposals related to the dapp's business logic
+  TOPIC_APPLICATION_BUSINESS_LOGIC = 4;
+  // Proposals related to governance
+  TOPIC_GOVERNANCE = 5;
+  // Proposals related to treasury management
+  TOPIC_TREASURY_ASSET_MANAGEMENT = 6;
+  // Critical proposals related to dapp operations
+  TOPIC_CRITICAL_DAPP_OPERATIONS = 7;
```

Comment:
for now, prevent registering custom proposal types under TOPIC_CRITICAL_DAPP_OPERATIONS and TOPIC_TREASURY_ASSET_MANAGEMENT